buildSrc/src/main/groovy/org/elasticsearch/gradle/NoticeTask.groovy

```diff
@@ -49,25 +55,38 @@ public class NoticeTask extends DefaultTask {
         dependencies.add(project)
     }
 
+    /** Add notices from an extra directory. */
+    public void extraLicensesDir(File extraLicensesDir) {
```

Comment:
I don't think we need to have a separation of the original and "extra" licenses dir.  We should just have a list of directories to iterate.